features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/statistics/usage.jag

```diff
@@ -490,4 +501,125 @@ function getAPIsFromAPIRequestsPerHourTable(from,to){
     }
 }
 
+
+function getAppsForThrottleStats(providerName, apiName) {
+    var usage = [];
+    var usageout = [];
+    try {
+        usage = usageClient.getAppsForThrottleStats(providerName,apiName);
+        if (log.isDebugEnabled()) {
+            log.debug("getProviderAPIFirstAccessTime for : ");
+        }
+        if (usage == null) {
+            return {
+                error:true
+            };
+
+        } else {
+            usage=JSON.parse(gson.toJson(usage));
+            return {
+                error:false,
+                usage:usage
+            };
+        }
+    } catch (e) {
+        log.error(e.message);
+        return {
+            error:e,
+            usage:null
+
+        };
+    }
+}
+
+function getThrottleDataOfApplication(appName,providerName,fromDate, toDate) {
+    var usage = [];
+    var usageout = [];
+    try {
+        usage = usageClient.getThrottleDataOfApplication(appName,providerName,fromDate, toDate);
+        if (log.isDebugEnabled()) {
+            log.debug("getProviderAPIFirstAccessTime for : ");
+        }
+        if (usage == null) {
+            return {
+                error:true
+            };
+
+        } else {
+            usage=JSON.parse(gson.toJson(usage));
+            return {
+                error:false,
+                usage:usage
+            };
+        }
+    } catch (e) {
+        log.error(e.message);
+        return {
+            error:e,
+            usage:null
+
+        };
+    }
+}
+
+
+function getAPIsForThrottleStats(providerName) {
+    var usage = [];
+    var usageout = [];
+    try {
+        usage = usageClient.getAPIsForThrottleStats(providerName);
+        if (log.isDebugEnabled()) {
+            log.debug("getProviderAPIFirstAccessTime for : ");
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea